binary_search_tree/binary_search_tree.py

```diff
@@ -8,62 +10,255 @@
+        # left --> root --> right
+        if self.left is not None:
+        # recursive case? closest to the base case
+            self.left.in_order_print()
+        print(self.value) # from BST node
+        
+        if self.right is not None:
+            self.right.in_order_print()
+        # build up my call stack to see what happens
+        
+    # BFT: You first explore all the nodes one step away, then all the nodes two steps away, etc.
+    # Level 1 (node level) first, and then level 2 (child node level) second, etc.
+    
     # Print the value of every node, starting with the given node,
-    # in an iterative breadth first traversal
-    def bft_print(self):
-        pass
-
+    # in an iterative Breadth First Traversal - we are not working with recursion - How about QUEUE? First In, First Out (FIFO) data structure.
+    def bft_print(self): # this function will only get called once
+        # create a queue for nodes
+        qq = deque()
+        # add the first node to the queue
+        qq.append(self)
+        
+        # while queue is not empty
+        while len(qq) > 0:
+            # remove the first node from the queue
+            current = qq.popleft()
+            # print the removed node
+            print(current.value)
+            
+            # add all children into the queue
+            if current.left:
+                qq.append(current.left)
+            if current.right:
+                qq.append(current.right)
+    
     # Print the value of every node, starting with the given node,
-    # in an iterative depth first traversal
+    # in an iterative depth first traversal (DFT) - not recursion
     def dft_print(self):
-        pass
+        # create a stack for nodes
+        s = []
+        # add the first node to the stack
+        s.append(self)
+        
+        # while the stack is not empty
+        while len(s) > 0:
+            # get the current node from the top of the stack
+            current = s.pop()
+            # print that node
+            print(current.value)
+            # add all children to the stack
+            # keep in mind, the order you add to the children, will matter
+            if current.left:
+                s.append(current.left)
+            if current.right:
+                s.append(current.right)
+                
 
     # Stretch Goals -------------------------
     # Note: Research may be required
 
     # Print Pre-order recursive DFT
-    def pre_order_dft(self):
-        pass
+    # def pre_order_dft(self):
+    #     pass
 
     # Print Post-order recursive DFT
-    def post_order_dft(self):
-        pass
+    # def post_order_dft(self):
+    #     pass
+    
+root_node = BSTNode(8) # no insert function needs to be called
+root_node.insert(3)
+root_node.insert(10)
+root_node.insert(9)
+root_node.insert(12)
+
+# JS: const print_node = (x) => { console.log(x) }
+print_node = lambda x: print(f'current_node is: {x}')
```

Comment:
These lambda functions in Python, are examples of _anonymous functions_, which are super useful for when you need small helper functions, often inside of other functions. And, our school is named after them.